lib/src/migrators/module.dart

```diff
@@ -870,23 +878,52 @@ class _ModuleMigrationVisitor extends MigrationVisitor {
       if (expectedUrl != url) continue;
 
       var newName = renamedMembers[declaration] ?? declaration.name;
-      if (declaration.member is VariableDeclaration) newName = "\$$newName";
+      String importOnlyPrefix;
+      if (declaration.isImportOnly && declaration.forward.prefix != null) {
+        importOnlyPrefix = declaration.forward.prefix;
+        newName = declaration.name.substring(importOnlyPrefix.length);
+      }
+      var formattedNewName =
+          declaration.member is VariableDeclaration ? '\$$newName' : newName;
       if (_shouldForward(declaration.name) &&
           !declaration.name.startsWith('-')) {
-        shown.add(newName);
-      } else if (!newName.startsWith('-') && !newName.startsWith(r'$-')) {
-        hidden.add(newName);
+        var subprefix = "";
+        if (prefixToRemove != null &&
+            declaration.name.startsWith(prefixToRemove) &&
+            importOnlyPrefix != null) {
+          subprefix = importOnlyPrefix.substring(prefixToRemove.length);
+        }
+        if (declaration.name.length != newName.length) _needsImportOnly = true;
+        shownByPrefix.putIfAbsent(subprefix, () => {}).add(formattedNewName);
+      } else if (!newName.startsWith('-')) {
+        hidden.add(formattedNewName);
       }
     }
-    if (shown.isEmpty) return null;
-
+    if (shownByPrefix.isEmpty) return null;
     var tuple = _absoluteUrlToDependency(url);
-    var forward = '@forward "${tuple.item1}"';
-    if (hidden.isNotEmpty) {
-      var hiddenMembers = hidden.toList()..sort();
-      forward += ' hide ${hiddenMembers.join(", ")}';
+    var forwardBase = '@forward "${tuple.item1}"';
+    var forwards = <String>[];
+    for (var subprefix in shownByPrefix.keys.toList()..sort()) {
+      var allHidden = {
+        ...hidden,
+        for (var other in shownByPrefix.keys)
+          if (other != subprefix) ...shownByPrefix[other]
+      }.toList()
+        ..sort();
+      var forward = forwardBase;
+      if (subprefix.isNotEmpty) {
+        forward += ' as $subprefix*';
+        allHidden = [
+          for (var item in allHidden)
+            item.startsWith(r'$')
+                ? '\$$subprefix${item.substring(1)}'
+                : '$subprefix$item'
+        ];
+      }
+      if (allHidden.isNotEmpty) forward += ' hide ${allHidden.join(", ")}';
```

Comment:
This generation logic is complex... are there test cases for generating multiple forwards with different `hide`s and prefixes?

Reply:
Added a test for different subprefixes and fixed a bug in `References` that it revealed.